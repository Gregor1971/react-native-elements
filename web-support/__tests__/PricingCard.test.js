import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../samples';
import { PricingCard as Component } from '../../src';

const props = {
  'required props: title, price, color, button, & info': {
    component: Component,
    props: {
      title: 'title',
      price: 'price',
      color: '#071',
      button: {
        title: 'buttonTitle',
        icon: 'pets',
        buttonStyle: { borderWidth: 2 },
      },
      info: ['info0', 'info1'],
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'onButtonPress & required props': {
    component: Component,
    props: {
      title: 'title',
      price: 'price',
      color: '#071',
      button: {
        title: 'buttonTitle',
        icon: 'pets',
        buttonStyle: { borderWidth: 2 },
      },
      info: ['info0', 'info1'],
      onButtonPress: () => {
        console.log('button pressed');
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'containerStyle & required props': {
    component: Component,
    props: {
      title: 'title',
      price: 'price',
      color: '#071',
      button: {
        title: 'buttonTitle',
        icon: 'pets',
        buttonStyle: { borderWidth: 2 },
      },
      info: ['info0', 'info1'],
      containerStyle: { borderWidth: 5 },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'wrapperStyle & required props': {
    component: Component,
    props: {
      title: 'title',
      price: 'price',
      color: '#071',
      button: {
        title: 'buttonTitle',
        icon: 'pets',
        buttonStyle: { borderWidth: 2 },
      },
      info: ['info0', 'info1'],
      wrapperStyle: { borderWidth: 5 },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'titleFont & required props': {
    component: Component,
    props: {
      title: 'title',
      price: 'price',
      color: '#071',
      button: {
        title: 'buttonTitle',
        icon: 'pets',
        buttonStyle: { borderWidth: 2 },
      },
      info: ['info0', 'info1'],
      titleFont: 'Courier New',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'pricingFont & required props': {
    component: Component,
    props: {
      title: 'title',
      price: 'price',
      color: '#071',
      button: {
        title: 'buttonTitle',
        icon: 'pets',
        buttonStyle: { borderWidth: 2 },
      },
      info: ['info0', 'info1'],
      pricingFont: 'Courier New',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'infoFont & required props': {
    component: Component,
    props: {
      title: 'title',
      price: 'price',
      color: '#071',
      button: {
        title: 'buttonTitle',
        icon: 'pets',
        buttonStyle: { borderWidth: 2 },
      },
      info: ['info0', 'info1'],
      infoFont: 'Courier New',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'buttonFont & required props': {
    component: Component,
    props: {
      title: 'title',
      price: 'price',
      color: '#071',
      button: {
        title: 'buttonTitle',
        icon: 'pets',
        buttonStyle: { borderWidth: 2 },
      },
      info: ['info0', 'info1'],
      buttonFont: 'Courier New',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
};

const samples = {
  props,
};

parseSamples(
  {
    Cards: { sectionComponents: { PricingCard: { samples } } },
  },
  options
);
