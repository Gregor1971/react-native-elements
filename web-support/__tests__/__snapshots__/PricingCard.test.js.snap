// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards: PricingCard: [ prop: buttonFont & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      134,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        135,
        undefined,
      ]
    }
  >
    <TextElement
      style={
        Array [
          136,
          Object {
            "color": "#071",
          },
          undefined,
        ]
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Array [
          137,
          undefined,
        ]
      }
    >
      price
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info0
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info1
    </TextElement>
    <Button
      buttonStyle={
        Array [
          139,
          Object {
            "borderWidth": 2,
          },
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      fontFamily="Courier New"
      icon={
        Object {
          "name": "pets",
        }
      }
      title="buttonTitle"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: [ prop: containerStyle & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      134,
      Object {
        "borderWidth": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        135,
        undefined,
      ]
    }
  >
    <TextElement
      style={
        Array [
          136,
          Object {
            "color": "#071",
          },
          undefined,
        ]
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Array [
          137,
          undefined,
        ]
      }
    >
      price
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info0
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info1
    </TextElement>
    <Button
      buttonStyle={
        Array [
          139,
          Object {
            "borderWidth": 2,
          },
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      icon={
        Object {
          "name": "pets",
        }
      }
      title="buttonTitle"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: [ prop: infoFont & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      134,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        135,
        undefined,
      ]
    }
  >
    <TextElement
      style={
        Array [
          136,
          Object {
            "color": "#071",
          },
          undefined,
        ]
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Array [
          137,
          undefined,
        ]
      }
    >
      price
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          Object {
            "fontFamily": "Courier New",
          },
        ]
      }
    >
      info0
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          Object {
            "fontFamily": "Courier New",
          },
        ]
      }
    >
      info1
    </TextElement>
    <Button
      buttonStyle={
        Array [
          139,
          Object {
            "borderWidth": 2,
          },
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      icon={
        Object {
          "name": "pets",
        }
      }
      title="buttonTitle"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: [ prop: onButtonPress & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      134,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        135,
        undefined,
      ]
    }
  >
    <TextElement
      style={
        Array [
          136,
          Object {
            "color": "#071",
          },
          undefined,
        ]
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Array [
          137,
          undefined,
        ]
      }
    >
      price
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info0
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info1
    </TextElement>
    <Button
      buttonStyle={
        Array [
          139,
          Object {
            "borderWidth": 2,
          },
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      icon={
        Object {
          "name": "pets",
        }
      }
      onPress={[Function]}
      title="buttonTitle"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: [ prop: pricingFont & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      134,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        135,
        undefined,
      ]
    }
  >
    <TextElement
      style={
        Array [
          136,
          Object {
            "color": "#071",
          },
          undefined,
        ]
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Array [
          137,
          Object {
            "fontFamily": "Courier New",
          },
        ]
      }
    >
      price
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info0
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info1
    </TextElement>
    <Button
      buttonStyle={
        Array [
          139,
          Object {
            "borderWidth": 2,
          },
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      icon={
        Object {
          "name": "pets",
        }
      }
      title="buttonTitle"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: [ prop: required props: title, price, color, button, & info ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      134,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        135,
        undefined,
      ]
    }
  >
    <TextElement
      style={
        Array [
          136,
          Object {
            "color": "#071",
          },
          undefined,
        ]
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Array [
          137,
          undefined,
        ]
      }
    >
      price
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info0
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info1
    </TextElement>
    <Button
      buttonStyle={
        Array [
          139,
          Object {
            "borderWidth": 2,
          },
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      icon={
        Object {
          "name": "pets",
        }
      }
      title="buttonTitle"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: [ prop: titleFont & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      134,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        135,
        undefined,
      ]
    }
  >
    <TextElement
      style={
        Array [
          136,
          Object {
            "color": "#071",
          },
          Object {
            "fontFamily": "Courier New",
          },
        ]
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Array [
          137,
          undefined,
        ]
      }
    >
      price
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info0
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info1
    </TextElement>
    <Button
      buttonStyle={
        Array [
          139,
          Object {
            "borderWidth": 2,
          },
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      icon={
        Object {
          "name": "pets",
        }
      }
      title="buttonTitle"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: [ prop: wrapperStyle & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      134,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        135,
        Object {
          "borderWidth": 5,
        },
      ]
    }
  >
    <TextElement
      style={
        Array [
          136,
          Object {
            "color": "#071",
          },
          undefined,
        ]
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Array [
          137,
          undefined,
        ]
      }
    >
      price
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info0
    </TextElement>
    <TextElement
      style={
        Array [
          138,
          undefined,
        ]
      }
    >
      info1
    </TextElement>
    <Button
      buttonStyle={
        Array [
          139,
          Object {
            "borderWidth": 2,
          },
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      icon={
        Object {
          "name": "pets",
        }
      }
      title="buttonTitle"
    />
  </View>
</View>
`;
